core/src/main/java/com/sequenceiq/cloudbreak/converter/ImageCatalogToResponseConverter.java

```diff
@@ -1,20 +1,35 @@
 package com.sequenceiq.cloudbreak.converter;
 
+import javax.inject.Inject;
+
 import org.springframework.stereotype.Component;
 
 import com.sequenceiq.cloudbreak.api.model.imagecatalog.ImageCatalogResponse;
 import com.sequenceiq.cloudbreak.domain.ImageCatalog;
+import com.sequenceiq.cloudbreak.service.image.ImageCatalogService;
 
 @Component
 public class ImageCatalogToResponseConverter extends AbstractConversionServiceAwareConverter<ImageCatalog, ImageCatalogResponse> {
 
+    @Inject
+    private ImageCatalogService imageCatalogService;
+
     @Override
     public ImageCatalogResponse convert(ImageCatalog source) {
         ImageCatalogResponse imageCatalogResponse = new ImageCatalogResponse();
         imageCatalogResponse.setId(source.getId());
         imageCatalogResponse.setPublicInAccount(source.isPublicInAccount());
-        imageCatalogResponse.setName(source.getImageCatalogName());
         imageCatalogResponse.setUrl(source.getImageCatalogUrl());
+
+        String imageCatalogName = source.getImageCatalogName();
+        imageCatalogResponse.setDefault(isDefault(imageCatalogName));
+        imageCatalogResponse.setName(imageCatalogName);
+
         return imageCatalogResponse;
     }
+
+    private boolean isDefault(String imageCatalogName) {
+        String defaultImageCatalogName = imageCatalogService.getDefaultImageCatalogName();
+        return imageCatalogName.equals(defaultImageCatalogName) || (imageCatalogService.isEnvDefault(imageCatalogName) && defaultImageCatalogName == null);
```

Comment:
defaultImageCatalogName == null is light weighter than service call should be the first